### PR TITLE
fixes and improvements for tidal filter

### PR DESCRIPTION
**Pull request by kthyng**
The tidal filter isn't working correctly at least in certain cases, but this fixes it and expands some options.

### CODE REVIEW

**Comment by kthyng** [comment]
@ocefpaf Can you take a look at this PR?

File: oceans/filters.py [hunk 1]
```diff
@@ -608,14 +605,26 @@ def pl33tn(x, dt=1.0, T=33.0, mode="valid"):
     """
 
     import cf_xarray  # noqa: F401
+    import pandas as pd
     import xarray as xr
 
     if isinstance(x, xr.Dataset):
         raise TypeError("Input a DataArray not a Dataset.")
+    elif isinstance(x, pd.DataFrame):
+        raise TypeError("Input a Series not a DataFrame.")
```
Comment (ocefpaf) [inline oceans/filters.py:614]: Let's do a single check for both:

```python
isinstance(x, (xr.DataArray, pd.DataFrame))
```

Comment (kthyng) [reply oceans/filters.py]: see new commits

File: oceans/filters.py [hunk 2]
```diff
@@ -621,7 +621,7 @@ def pl33tn(x, dt=1.0, T=33.0, mode="valid", t=None):
 
     # find dt in units of hours
     if isinstance(x, xr.DataArray):
-        dt = (x.cf["T"][1] - x.cf["T"][0]) / np.timedelta64(int(3.6e12))
+        dt = (x.cf["T"][1] - x.cf["T"][0]) / np.timedelta64(360_000_0000_000)  # This is a bit more readable.
```
[the remark this answers is not in the document]
Comment (kthyng) [reply oceans/filters.py]: Is that number supposed to have 4 zeros in a row? Haven't seen this syntax before.

[the remark this answers is not in the document]
Comment (ocefpaf) [reply oceans/filters.py]: Nope. It should be 3.

File: oceans/filters.py [hunk 3]
```diff
@@ -608,14 +605,25 @@ def pl33tn(x, dt=1.0, T=33.0, mode="valid"):
...
+        pd.core.indexes.datetimes.DatetimeIndex,
+    ):
+        raise TypeError("Input Series needs to have parsed datetime indices.")
+
+    # find dt in units of hours
     if isinstance(x, xr.DataArray):
-        # find dt in units of hours
-        dt = (x.cf["T"][1] - x.cf["T"][0]) * 1e-9 / 3600
+        dt = (x.cf["T"][1] - x.cf["T"][0]) / np.timedelta64(
+            360_000_0000_000,
```
Comment (ocefpaf) [inline oceans/filters.py:623]: This is the same as 360000000000 but more readable.
```suggestion
            360_000_000_000,
```

**Comment by kthyng** [comment]
@ocefpaf Can I merge this? And could you release a new version afterward?

**Comment by kthyng** [comment]
@ocefpaf I'm also happy to do the release myself!

**Pull request merged**